backend/core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/CorsFilter.java

```diff
@@ -44,6 +44,7 @@ public void handle(ClassicHttpRequest request, HttpFilterChain.ResponseTrigger r
       }
       response.addHeader("Access-Control-Allow-Methods", "GET, POST, OPTIONS");
       response.addHeader("Access-Control-Allow-Private-Network", true);
+      response.addHeader("Access-Control-Allow-Headers", "baggage, sentry-trace");
```

Comment:
Unlike SQ, SC uses a third-party library called sentry. This library put these [headers](https://github.com/contentful/contentful.js/issues/1381) to the cors request that causes this error:

> Access to fetch at ... from ... origin has been blocked by CORS policy: Request header field baggage is not allowed by Access-Control-Allow-Headers in preflight response.

One solution to fix this issue is to allow them in the response (current solution, on SLCORE side). Another one would be [disabling](https://github.com/getsentry/sentry-javascript/issues/6077#issuecomment-1665740420) for fetching in the sentry config. The last one would be [enabling](https://github.com/getsentry/sentry-javascript/issues/6077#issuecomment-1667307974) these headers only for given servers.

Reply:
I'll add a comment about it

Reply:
I think it would make more sense to disable those headers for local requests in the first place than patch the SL side. But I'm not familiar with Sentry so not sure if this is hard to do or not

Reply:
Ok, I managed to fix the issue on the SC side. I'll remove this code from here